R/find_bad_apples.R

```diff
@@ -2,23 +2,98 @@
 #'
 #' @description
 #' This function uses a univariate approach to outlier detection.
-#' For each column with outliers (values that are 3 or more standard deviations from the mean),
+#' For each column with outliers (values that are 2 or more standard deviations from the mean),
 #' this function will create a reference list of row indices with outliers, and
 #' the total number of outliers in that column.
 #'
 #' Note: This function works best for small datasets with unimodal variable distributions.
 #'
 #' @param df A dataframe containing numeric data
 #'
-#' @return A dataframe with columns for 'Variable' (dataframe column name),
-#' 'Indices' (list of row indices with outliers),
-#' 'Total Outliers' (number of outliers in the column)
+#' @return A dataframe with columns for 'variable' (dataframe column name),
+#' 'total_outliers' (number of outliers in the column), and
+#' 'indices' (list of row indices with outliers)
 #'
 #' @examples
 #' find_bad_apples(df)
 #'
 #' @export
 
 find_bad_apples <- function(df) {
-  # insert code here
+
+    # Returns a warning and stops the function if the dataframe includes non-numeric values
+    nums <- unlist(lapply(df, is.numeric))
+    if (dim(df)[2] != sum(nums)){
+        warning('Every column in your dataframe must be numeric.')
+        break
+
+        # Returns a warning and stops the function if the dataframe has fewer than 30 rows
+    } else {
+        if (dim(df)[1] < 30){
+            warning('Sorry, you don\'t have enough data. The dataframe needs to have at least 30 rows.')
+            break
+
+        } else {
+            # Creates empty output dataframe that will be added to
+            output <- tibble(variable = character(),
+                             indices = tibble(),
+                             total_outliers = integer()) %>%
+                group_by(variable, total_outliers) %>%
+                nest()
```

Comment:
It is best practise when writing R functions to specify what package a function came from. The code also will not work in some cases if you don't do this. See here for reference: https://r-pkgs.org/namespace.html#import-r

Note I didn't test the suggestion below. This is just what I think it should look like.

```
            output <- dplyr::tibble(variable = character(),
                             indices = dplyr::tibble(),
                             total_outliers = integer()) %>%
                dplyr::group_by(variable, total_outliers) %>%
                dplyr::nest()
```

Reply:
Thanks, @SamEdwardes ! I've made the changes in the new update that I'll make a pull request for soon.